src/ConfigurationTransform/Services/Extensions/ProjectItemExtensions.cs

```diff
@@ -38,19 +39,43 @@ public static string GetFullPath(this ProjectItem source)
 
         public static ProjectItem ParentProjectItemOrDefault(this ProjectItem source)
         {
-            var containingProject = source.ContainingProject;
-            if (containingProject == null) return null;
+            ProjectItem parent = null;
 
-            foreach (var projectItem in containingProject.ProjectItems.AsEnumerable())
+            // trying to convert source.Collection.Parent into a ProjectItem since this is the only type that
+            // makes sense in when wanting to show a preview of a transform.
+            try
             {
-                var found = projectItem.ContainingProjectItem(item => item.Name == source.Name).FirstOrDefault();
-                // found self(source is root)?
-                if (projectItem == found) return null;
-                // found sub project item?
-                if (found != null) return projectItem;
+                // getting the parent of the clicked project item
+                parent = (ProjectItem)source.Collection.Parent;
+            }
+            catch (InvalidCastException e)
+            {
+                // returning null to show info/error-message.
+                return null;
+            }
+
+            if (source.IsFolder())
+            {
```

Comment:
i think it's redundant leftovers

Reply:
=D I think my idea was to avoid cases where the tranformation config was not nested in a "parent config" file but your solution is way better and works when the parent i a project as well =D